### ISSUE TITLE
Anyway to silence 'WARN EsInputFormat: Cannot determine task id...'

### ISSUE THREAD

**Issue by royrusso** [elastic/elasticsearch-hadoop#427]
Looking at HadoopCfgUtils, it looks like this is a cfg setting, but setting it in the RDD Config, doesn't seem to have an effect. Is there somewhere else to set this?

es_conf_1 = {
    "es.nodes": "localhost",
    "es.port": "9200",
    "es.resource.read": READ_PATH,
    "es.resource.write": WRITE_PATH,
    "mapreduce.task.id" : "SOME_TASK_ID"
}

**Comment by costin** [comment]
`mapreduce.task.id` is not an Elastic configuration but rather a a Hadoop generated one. It looks like you are using Spark through the Map/Reduce layer and I'm afraid, since the environment doesn't set it up, there's nothing the connector can use.
If you turn on logging to the TRACE level, besides the warning one should get also the Hadoop configuration available which would be useful to inspect (please post it as a gist).
Otherwise you can use the native integration which doesn't have this issue.

**Comment by royrusso** [comment]
I believe the native integration is well-documented for Scala/Java, but this is being developed using python on pyspark. Looking through the codebase and docs, I didn't see native integration using python. Did I miss it, by chance?

**Comment by costin** [comment]
No, there's no python integration. Assuming you are using the latest connector version (the dev one), do [enable](http://www.elastic.co/guide/en/elasticsearch/hadoop/master/logging.html) logging and post a gist with the logging for your configuration which should occur right after your WARN message.

**Comment by royrusso** [comment]
Thanks!

Gist: https://gist.github.com/royrusso/f75aa9d8e781410c449c

**Comment by costin** [comment]
@royrusso Indeed, your configuration does not contain any of the needed Map/Reduce running information. That is, the Hadoop `InputFormat`/`OutputFormat` are called but the environment doesn't contain any runtime configuration hence why you get the warning message.
A while back I've raised an [issue](https://issues.apache.org/jira/browse/SPARK-1128) about this which was fixed in Spark 1.0 however only on the Scala/Java side. If you look at the contribution, you'll notice the required properties are added before the tasks are run however as you seem to be using Python, there's a high chance that method is not invoked and as such the properties are not properly set.

In other words, it seems to be a bug in Spark so I would raise this with the team along with a small snippet of code - the fix should be fairly straight forward. There also might be other Hadoop properties that are not properly set.

In the mean time, simply ignore the `WARN`ing or, if you insight, turn it off by using a warning level of `ERROR` on the said logger (though you might be losing some other important messages).
There's nothing else that I can see us do - note that despite the warning, the connector performs a best attempt to do the right thing.

Hope this helps,

**Comment by royrusso** [comment]
No worries. For now, I'll silence the noise and report it to the Spark team. Thanks for the help!

**Comment by costin** [comment]
No problem. I'll close this then - once you get a resolution (or created the report in the Spark tracker) please post a reference here; it might help others bumping into the same issue.

Cheers,

**Comment by SergeVil** [comment]
I have the same problem now. What information I can provide you for the fix?

**Comment by jbaiera** [comment]
@SergeVil as denoted in the above thread, this is an issue with Spark

**Comment by SergeVil** [comment]
Thank you. Do you have an idea where I can submit it? Is it any collaboration between the teams?

**Comment by jbaiera** [comment]
You will have to search the Spark issues, there is no official collaboration between this project and Spark.
